### PR TITLE
feat(signer): Use structures for all the new request and response types

### PR DESCRIPTION
**Pull request by bitdivine**
# Motivation
The signing API will have to be backwards compatible for a long time.  Passing structures for request and response types allows adding optional fields without breaking the APPI.

# Changes
- Use structures for all new paid API request and response types.
- Homogenize the paid API endpoints.

# Tests
Existing CI should suffice

### CODE REVIEW

File: src/signer/api/src/types/eth.rs
Comment (bitdivine) [inline src/signer/api/src/types/eth.rs:1]: For each paid Eth endpoint, we have Request, Response and Error types:

File: src/signer/api/src/types/generic.rs
Comment (bitdivine) [inline src/signer/api/src/types/generic.rs:1]: The generic endpoints have the same signatures as the canister signing endpoints, apart from supporting payment errors as well.  So we don't need additional types here.

File: src/signer/canister/signer.did [hunk 1]
```diff
@@ -188,17 +207,19 @@ service : (Arg) -> {
   btc_caller_send : (SendBtcRequest, opt PaymentType) -> (Result_2);
   caller_eth_address : () -> (text);
   config : () -> (Config) query;
+  eth_address : (EthAddressRequest, opt PaymentType) -> (Result_3);
   eth_address_of : (principal) -> (text);
-  eth_address_of_caller : (opt PaymentType) -> (Result_3);
```
[the remark this answers is not in the document]
Comment (bitdivine) [reply src/signer/canister/signer.did:192]: Done

File: src/signer/canister/src/lib.rs [hunk 2]
```diff
@@ -136,81 +150,89 @@ async fn generic_sign_with_ecdsa(
...
 // ////////////////////
 // // ETHEREUM UTILS //
 // ////////////////////
 
-/// Returns the Ethereum address of the caller.
+/// Returns the Ethereum address of a specified user.
+///
+/// If no user is specified, the caller's address is returned.
 #[update(guard = "caller_is_not_anonymous")]
-async fn eth_address_of_caller(
```
[the remark this answers is not in the document]
Comment (lmuntaner) [reply src/signer/canister/src/lib.rs:146]: Sounds good for me!

**Review by lmuntaner — APPROVED** [review]
LGTM! Thanks

As discussed, the endpoints changed in this PR are not yet being used in Oisy.

**Pull request merged**